### PR TITLE
[V3] feat: remove auto-connect functionality

### PR DESCRIPTION
**Pull request by cartertinney**
* Removed `AutoConnectStage` from the pipeline
* Removed `auto_connect` kwarg from factory methods
* Updated DPS `.register()` to manually do a connect in lieu of auto-connect functionality
* Updated tests accordingly (except regression/stress)

### CODE REVIEW

File: azure-iot-device/azure/iot/device/iothub/sync_clients.py [hunk 1]
```diff
@@ -289,23 +294,14 @@ def send_message(self, message):
...
         :type message: :class:`azure.iot.device.Message` or str
 
-        :raises: :class:`azure.iot.device.exceptions.CredentialError` if credentials are invalid
-            and a connection cannot be established.
-        :raises: :class:`azure.iot.device.exceptions.ConnectionFailedError` if a establishing a
-            connection results in failure.
-        :raises: :class:`azure.iot.device.exceptions.ConnectionDroppedError` if connection is lost
-            during execution.
-        :raises: :class:`azure.iot.device.exceptions.OperationTimeout` if connection attempt
-            times out
```
Comment (BertKleewein) [inline azure-iot-device/azure/iot/device/iothub/sync_clients.py:306]: What do I get if I can make send_message raise these exceptions?

Comment (cartertinney) [reply azure-iot-device/azure/iot/device/iothub/sync_clients.py:306]: I bet you $100 you can't make it raise OperationTimeout.

Not playing the ConnectionDroppedError odds though

**Review by BertKleewein — APPROVED** [review]
:shipit:

**Pull request merged**